extension/goplugin/database.go

```diff
@@ -51,17 +52,29 @@ func (db *Database) Clone() *Database {
 	}
 }
 
+func isWithin() bool {
+	pc, _, _, ok := runtime.Caller(4)
+	if !ok {
+		panic("runtime.Caller failed")
+	}
+	f := runtime.FuncForPC(pc)
+	if f == nil {
+		panic("runtime.FuncForPC failed")
+	}
+	return f.Name() == "github.com/cloudwan/gohan/extension/goext.within"
+}
+
 // Begin starts a new transaction
 func (db *Database) Begin() (goext.ITransaction, error) {
 	t, _ := db.raw.Begin()
-	return &Transaction{t}, nil
+	return &Transaction{t, isWithin()}, nil
```

Comment:
If Begin() shouldn't be called directly, it should be a private function (and we wouldn't need to add any runtime checks).
On the other hand, if some plugins really need transactions not started by Within(), we'll get a lot of spurious erros. If we want to keep the functionality, let's call it BeginUnsafe()